projects/igniteui-angular/src/lib/calendar/calendar-base.ts

```diff
@@ -258,6 +259,9 @@ export class IgxCalendarBaseDirective implements ControlValueAccessor {
      */
     public set locale(value: string) {
         this._locale = value;
+        if (!this.weekStart) {
```

Comment:
If we have set `weekStart` to be WEEKDAYS.Sunday (equals 0), this check will always evaluate as `true` and override the value. Example:
```html
<igx-date-picker locale="FR" [weekStart]="0" 
```
`weekStart` will always be 1, which is the default of french locale.